AndroidAnnotations/androidannotations-core/androidannotations/src/main/java/org/androidannotations/helper/ValidatorHelper.java

```diff
@@ -546,8 +546,10 @@ public void canBePutInABundle(Element element, ElementValidation valid) {
 			if (typeMirror.getKind() != TypeKind.NONE) {
 				TypeMirror parcelableType = annotationHelper.typeElementFromQualifiedName(CanonicalNameConstants.PARCELABLE).asType();
 				TypeMirror serializableType = annotationHelper.typeElementFromQualifiedName("java.io.Serializable").asType();
-				if (!annotationHelper.isSubtype(typeMirror, parcelableType) && !annotationHelper.isSubtype(typeMirror, serializableType)) {
-					valid.addError("Unrecognized type. Please let your attribute be primitive or implement Serializable or Parcelable");
+				if (!annotationHelper.isSubtype(typeMirror, parcelableType)
+						&& !annotationHelper.isSubtype(typeMirror, serializableType)
+						&& !annotationHelper.isParcelType(typeMirror)) {
```

Comment:
I just realized we have a problem here. We only allow `Parcelable`, `Serializable` (and now Parceler supported) types. But that will leave out most of [these types for example](https://github.com/johncarl81/androidannotations/blob/parceler/AndroidAnnotations/androidannotations-core/androidannotations/src/main/java/org/androidannotations/helper/BundleHelper.java#L47). We have to fix this later...